service/controller/legacy/v25/resource/tccp/create.go

```diff
@@ -318,7 +318,10 @@ func (r *Resource) newTemplateBody(ctx context.Context, cr v1alpha1.AWSConfig, t
 				VersionBundleVersion: key.VersionBundleVersion(cr),
 			},
 			TenantClusterAccountID: cc.Status.TenantCluster.AWSAccountID,
-			TenantClusterKMSKeyARN: cc.Status.TenantCluster.Encryption.Key,
+		}
+
+		if r.encrypterBackend == encrypter.KMSBackend {
+			c.TenantClusterKMSKeyARN = cc.Status.TenantCluster.Encryption.Key
 		}
```

Comment:
I don't have a better solution for this. I just checked other conditions towards the backend types and they are all handled in the resource. Good enough for now.